mobly/controllers/android_device.py

```diff
@@ -366,7 +366,7 @@ def __init__(self, serial="", host_port=None, device_port=8080):
         self.adb_logcat_file_path = None
         self.adb = adb.AdbProxy(serial)
         self.fastboot = fastboot.FastbootProxy(serial)
-        if not self.is_bootloader:
+        if not self.is_bootloader and self.is_rootable:
```

Comment:
That's intentionally left there.
Since you brought up the point (during internal commit) that sometimes users do want to force adb root on a device regardless of the build, and we decided not to block it if user calls ad.root_adb intentionally.